libstuff/SQResult.cpp

```diff
@@ -183,3 +188,55 @@ SQResult& SQResult::operator=(const SQResult& other) {
     }
     return *this;
 }
+
+vector<SQResultRow>::iterator SQResult::begin() {
+    return rows.begin();
+}
+
+vector<SQResultRow>::iterator SQResult::end() {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::begin() const {
+    return rows.begin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::end() const {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cbegin() const {
+    return rows.cbegin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cend() const {
+    return rows.cend();
+}
+
+void SQResult::resize(size_t newSize) {
+    rows.resize(newSize);
+    for (auto& row : rows) {
+        row.result = this;
+    }
+}
```

Comment:
I would use constructor `(2)` here:
https://en.cppreference.com/w/cpp/container/vector/resize

And if we're increasing the size of `rows`, then pass an initializing element, something like:

```
void SQResult::resize(size_t newSize) {
    if (newSize > rows.size()) {
        rows.resize(newSize, SQResultRow(this));
    } else {
        rows.resize(newSize);
    }
}
```

This avoids reassigning `result` on all of the existing rows.